eng/pipelines/pull-request-common.yml

```diff
@@ -74,11 +68,12 @@ steps:
   - script: node eng/scripts/check-for-changed-files.js
     displayName: Check Git Status For Changed Files
 
-  - script: rush cspell
+  - script: node common/scripts/install-run-rush.js cspell
     displayName: Spell check
     condition: ne(variables['Agent.OS'], 'Windows_NT')
 
-  - script: rush e2e
+  - script: node common/scripts/install-run-rush.js e2e
     displayName: E2E Tests
-    # Skipping e2e test on windows node 16.x as there is a bug with npm and it doesn't seem to want to update correctly.
-    condition: ne(variables['Agent.OS'], 'Windows_NT')
+    # Skipping the e2e test on the prepare publish branch as there is a bug in the npm version installed on node 16
+    # causing npx to try to download the unpublished version of the cli.
+    condition: not(startsWith(variables['Build.SourceBranch'], 'refs/publish/'))
```

Comment:
Is it worth making this, "not publish AND node < 18?"

Reply:
yeah could be good